src/test-table.md

```diff
@@ -1,7 +1,11 @@
 # Test Tables
 
-Use table-driven tests with [subtests] to avoid duplicating code when the core
-test logic is repetitive.
+Table-driven tests with [subtests] can be a helpful pattern for writing tests
+to avoid duplicating code when the core test logic is repetitive.
+
+If a system under test needs to be tested against _multiple conditions_ where
+*only* the **inputs** and **outputs** change, a table-driven test *can* and **should**
+be used to reduce redundancy and improve readability.
```

Comment:
couple of nits:
 * I think we can tone it down a little (e.g., there's a lot of emphasis *only*, *can*, *should*)
 * I think it's useful to provide some context for why